plugins/optimization-detective/readme.txt

```diff
@@ -21,11 +23,18 @@ In order to increase the accuracy of identifying the LCP element, including acro
 
 = Technical Foundation =
 
-At the core of Optimization Detective is the “URL Metric”, information about a page according to how it was loaded by a client with a specific viewport width. This includes which elements were visible in the initial viewport and which one was the LCP element. Each URL on a site can have an associated set of these URL Metrics (stored in a custom post type) which are gathered from real users. It gathers a sample of URL Metrics according to common responsive breakpoints (e.g. mobile, tablet, and desktop). When no more URL Metrics are needed for a URL due to the sample size being obtained for the breakpoints, it discontinues serving the JavaScript to gather the metrics (leveraging the [web-vitals.js](https://github.com/GoogleChrome/web-vitals) library). With the URL Metrics in hand, the output-buffered page is sent through the HTML Tag Processor and--when the [Image Prioritizer](https://wordpress.org/plugins/image-prioritizer/) dependent plugin is installed--the images which were the LCP element for various breakpoints will get prioritized with high-priority preload links (along with `fetchpriority=high` on the actual `img` tag when it is the common LCP element across all breakpoints). LCP elements with background images added via inline `background-image` styles are also prioritized with preload links.
+At the core of Optimization Detective is the “URL Metric”, information about a page according to how it was loaded by a client with a specific viewport width. This includes which elements were visible in the initial viewport and which one was the LCP element. The URL Metric data is also extensible. Each URL on a site can have an associated set of these URL Metrics (stored in a custom post type) which are gathered from the visits of real users. It gathers samples of URL Metrics which are grouped according to WordPress's default responsive breakpoints:
+
+1. Mobile: 0-480px
+2. Phablet: 481-600px
```

Comment:
Never heard that word 😆 

Wondering, do we really need that breakpoint? Phablet and tablet are only ~120px and ~180px ranges respectively, which I find very little. Why not combine them like you have in the filter docs below? (Just a thought, that would need to be another PR though.)

Reply:
> Never heard that word 😆

It's actually a thing! Per [Wikipedia](https://en.wikipedia.org/wiki/Phablet) and in Google Search:

![image](https://github.com/user-attachments/assets/8817dc6e-82cb-48d0-92f9-44ea47556622)

> Wondering, do we really need that breakpoint? Phablet and tablet are only ~120px and ~180px ranges respectively, which I find very little. Why not combine them like you have in the filter docs below? (Just a thought, that would need to be another PR though.)

Good question. These numbers come from Gutenberg's breakpoints:

https://github.com/WordPress/gutenberg/blob/ab6e729daf775ee6065a48c210826af8281a0842/packages/base-styles/_breakpoints.scss#L11-L13

```
$break-medium: 782px;	// adminbar goes big
$break-small: 600px;
$break-mobile: 480px;
```

It's [used](https://github.com/WordPress/gutenberg/blob/ab6e729daf775ee6065a48c210826af8281a0842/packages/block-library/src/media-text/style.scss#L126-L145) in the Media & Text block, for example:

[Screen recording 2024-11-07 15.56.13.webm](https://github.com/user-attachments/assets/4d8e65a4-f403-4280-8fcf-e36fac835e11)

So this can be a critical breakpoint which affects the the LCP element.

Note also that this is the breakpoint at which the navigation menu transitions from mobile to non-mobile.

There are quite a few other uses of this `break-small` width: https://github.com/search?q=repo%3AWordPress%2Fgutenberg+%2F%5C%24break-small%7Cbreak-small%5C%28%5C%29%2F&type=code

I recall looking through the various breakpoints defined in Gutenberg and finding that these are the ones most commonly used in responsive styles. So that's why I picked these specifically.